main.cpp

```diff
@@ -76,19 +79,21 @@ struct PrintUpdated
     static float printUpdate(U* that,float* updatedValue )        //10
     {   
         if(updatedValue != nullptr)
```

Comment:
You can convert the function arguments to references.  You can't convert the return type.
- [ ] revise